### PR TITLE
Cage configuration tool

### PR DESCRIPTION
**Pull request by DonatJR**
~WIP: code to capture image from webcam is still missing but everything else can be reviewed~
__Edit__: webcam capture should now work correctly

image

example config generated (token string truncated):
```json
{
  "application_name": "Google Chrome",
  "application_path": "C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe",
  "has_signature": true,
  "binary_hash": "5399DDEE909C797705D4008EAFD471525FC1BE785B8C6DDDA27B6A5937305FF477CA8D250D6DD763EAEF033C0A4A13F76720234BE7F57CE585A57324B1C4243B",
  "token": "iVBORw0KGgoAAAANSUhEUgAAAoAAAAKOCAIAAAC5pT8EAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsQAAA7EAZUrDhsAAP+lSURBVHhezP1ldBxJvu6Nzpe77j17prvNLJZKzMzMJWZmlkpQolKJmVmWmRvc7WZmZm43M5jEKjHZvk/kX8quke3eM/vMed+j9axYkVFZqcrIiPjFExkZ+TdNgapAQ0VDXR...",
  "token_format": ".png",
  "additional_application": "None",
  "additional_application_path": "",
  "creation_date": 1528201183,
  "config_version": 1
}
```

### CODE REVIEW

**Comment by SailReal** [comment]
@DonatJR pls ping me as soon as the RP is ready to merge then I'll look into the code asap.

**Comment by DonatJR** [comment]
@SailReal ping :)

File: SharkCage/CageChooser/CageChooser.cs [hunk 1]
```diff
@@ -15,16 +15,20 @@ static class CageChooser
         static void Main()
         {
             // comment out '&& FALSE' to install service on application start
-#if DEBUG && FALSE
```
Comment (SailReal) [inline SharkCage/CageChooser/CageChooser.cs]: Is this now working without the &&? Maybe then remove the comment above?

Comment (DonatJR) [reply SharkCage/CageChooser/CageChooser.cs]: It was always working, the `&& FALSE` is just for when you don't want to start the install script every time. The comment is the explanation for that but could maybe be improved to explain both with and without `&& FALSE`

**Comment by SailReal** [comment]
Hmm I get an exception when I try to capture an image using the webcam and while storing the file, that's the config I used:
screen shot 2018-06-06 at 1 07 07 am

After saving the config, the following exception occurred: 
screen shot 2018-06-06 at 1 06 17 am

Tried it a few times with always the same result.

**Comment by DonatJR** [comment]
Should be fixed :+1:

**Comment by SailReal** [comment]
👍👍👍👍👍

**Pull request merged**